### ISSUE TITLE
[integration] [Bitbucket] wrong line number references

### ISSUE THREAD

**Issue by pjungermann** [backstage/backstage#9764]
## Expected Behavior

Line number references use the format 

- Bitbucket Cloud: `#lines-{start}[:{end}][,...]` (e.g., `#lines-188`, `#lines-188:208`, `#lines-7,21:42`)
- Bitbucket Server: `#{start}[-{end}][,...]` (e.g., `#188`, `#188-208`, `#7,21-42`)

## Actual Behavior

Line number format `#example.txt-42` is applied.

## Context

I discovered this while looking into the bitbucket integration.

E.g., used at the todo-backend plugin. Most places call `resolveUrl` without a lineNumber.

## Your Environment

- Output of `yarn backstage-cli info`: 

```text
yarn run v1.22.17
$ .../backstage/node_modules/.bin/backstage-cli info
OS:   Darwin 20.6.0 - darwin/x64
node: v16.3.0
yarn: 1.22.17
cli:  0.14.0 (installed)

Dependencies:
  @backstage/app-defaults                                  0.1.8
  @backstage/backend-common                                0.10.9
  @backstage/backend-tasks                                 0.1.8
  @backstage/catalog-client                                0.7.1
  @backstage/catalog-model                                 0.10.1
  @backstage/cli-common                                    0.1.8
  @backstage/cli                                           0.14.0
  @backstage/config-loader                                 0.9.5
  @backstage/config                                        0.1.15
  @backstage/core-app-api                                  0.5.3
  @backstage/core-components                               0.8.9
  @backstage/core-plugin-api                               0.6.1
  @backstage/errors                                        0.2.2
  @backstage/integration-react                             0.1.22
  @backstage/integration                                   0.7.4
  @backstage/plugin-api-docs                               0.7.3
  @backstage/plugin-app-backend                            0.3.26
  @backstage/plugin-auth-backend                           0.10.2
  @backstage/plugin-auth-node                              0.1.2
  @backstage/plugin-catalog-backend-module-msgraph         0.2.16
  @backstage/plugin-catalog-backend                        0.21.5
  @backstage/plugin-catalog-common                         0.1.4
  @backstage/plugin-catalog-graph                          0.2.11
  @backstage/plugin-catalog-import                         0.8.2
  @backstage/plugin-catalog-react                          0.6.15
  @backstage/plugin-catalog                                0.8.0
  @backstage/plugin-explore-react                          0.0.12
  @backstage/plugin-explore                                0.3.30
  @backstage/plugin-github-actions                         0.4.36
  @backstage/plugin-org                                    0.4.3
  @backstage/plugin-permission-common                      0.5.1
  @backstage/plugin-permission-node                        0.5.1
  @backstage/plugin-permission-react                       0.3.1
  @backstage/plugin-proxy-backend                          0.2.20
  @backstage/plugin-scaffolder-backend-module-cookiecutter 0.2.1
  @backstage/plugin-scaffolder-backend                     0.16.1
  @backstage/plugin-scaffolder-common                      0.2.1
  @backstage/plugin-scaffolder                             0.12.3
  @backstage/plugin-search-backend-node                    0.4.7
  @backstage/plugin-search-backend                         0.4.4
  @backstage/plugin-search                                 0.7.0
  @backstage/plugin-tech-radar                             0.5.6
  @backstage/plugin-techdocs-backend                       0.13.5
  @backstage/plugin-techdocs                               0.13.4
  @backstage/plugin-user-settings                          0.3.20
  @backstage/release-manifests                             0.0.2
  @backstage/search-common                                 0.2.4
  @backstage/techdocs-common                               0.11.9
  @backstage/test-utils                                    0.2.5
  @backstage/theme                                         0.2.15
  @backstage/types                                         0.1.3
  @backstage/version-bridge                                0.1.2
✨  Done in 2.07s.

```

**Comment by dhruvaldarji** [comment]
Bitbucket Server uses the below pattern to reference lines. 
The pattern starts with a `#` symbol and is followed by a comma-separated range of numbers. Each range can be represented by either just the `start` line number or the `start` and `end` (inclusive) line numbers separated by a `-`. 
```
#A,B,C-D,E
#1,5,10-20,35
```
For example, the above range defines line references in a file as line 1, line 5, an inclusive range from lines 10 to line 20, and line 35. 

The full URL pattern would be as below.
```
{BASE_URL}/projects/{PROJECT}/repos/{REPO}/browse/{PATH_TO_FILE}#1,5,10-20,35
```

Note, these are my findings per my internal company installation of Bitbucket Server v7.x.x.

**Comment by pjungermann** [comment]
@dhruvaldarji thank you for the input! I guess then the current implementation is wrong for both.

As of references to API docs in the code, it seem it was implemented against 7.9.0. But not sure about referencing lines.
